Week8AssignmentNew/ConsoleApp5/HomePage1.cs

```diff
@@ -0,0 +1,260 @@
+            Cart1 cart = new Cart1(driver);
+            while (pageCounter <= ppageCount)
+            {
+                int itemCount = driver.FindElements(By.XPath("(//img[contains(@alt,'Apple')])")).Count;
+                //int iitemsCount = DiplayTotalItemsPresent();
+                for (int index = 1; index <= itemCount; index++)
+                {
+                    bool itemUnavailable = ItemUnavailable(index);
+                    //bool itemAvailable = driver.FindElement(By.XPath(string.Format(UnavailableItem, index))).Displaye
+                    if (!itemUnavailable) // if item available
+                    {
+                        //add to cart.
+                        java.lang.Thread.sleep(1000);
+                        try
+                        {
+                            driver.FindElement(By.XPath(string.Format(availableitem, index))).Click();
+                        }
+                        catch (Exception e)
+                        {
+                            driver.Close();
+                        }
+                        java.lang.Thread.sleep(2000);
+
+                        //Window Handles implementation
+                        ArrayList tabs = new ArrayList(driver.WindowHandles);
+                        driver.SwitchTo().Window((string)tabs[1]);
+                        cart.AddItemToCart();
+                        java.lang.Thread.sleep(1000);
+                        driver.SwitchTo().Window(driver.WindowHandles.First());
+                    }
+                }
+                pageCounter++;
+                if (pageCounter <= ppageCount)
+                {
+                    ClickNextButton();
+                    Thread.Sleep(2000);
+                }                
+            }
+        }
+        public string FinalSearchResult()
+        {
+            string r = driver.FindElement(searchResult).Text;
+            return r;
+        }
+
+        public void ClickNextButton()
+        {
+            driver.FindElement(nextButton).Click();
+
+        }
+        public string GetProductName(int index)
+        {
+            string nameOfProduct = driver.FindElement(By.XPath(string.Format(productName, index))).Text;
+            return nameOfProduct;
+        }
+
+        public string GetProductPrice(int index)
+        {
+            string priceOfProduct = driver.FindElement(By.XPath(string.Format(productPrice, index))).Text;
+            return priceOfProduct;
+        }
+        public string TotalPriceOfProducts()
+        {
+            ArrayList tabs = new ArrayList(driver.WindowHandles);
+            driver.SwitchTo().Window((string)tabs[1]);
+            Thread.Sleep(2000);
+            string totalProductsCost = driver.FindElement(totalCost).Text;
+            return totalProductsCost;
+        }
+
+        //Verify Count of the product
+        public void FinalCountOfProduct()
+        {
+            string Result = FinalSearchResult();
```

Comment:
driver.FindElement(searchResult).Text; this can be done directly here since the page object as driver access.